src/containers/BooksForm.js

```diff
@@ -1,18 +1,75 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import { v1 as uuidv1 } from 'uuid';
+import PropTypes from 'prop-types';
+import { createBook } from '../actions/index';
 import CATEGORIES from '../constants/Categories';
 
-const BooksForm = () => (
-  <form>
-    <input type="text" value="" name="title" />
-    <select value="" name="category">
-      {
-          CATEGORIES.map(category => (
-            <option key={category} value={category}>{category}</option>
-          ))
-        }
-    </select>
-    <button type="submit">Submit</button>
-  </form>
-);
-
-export default BooksForm;
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: '',
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    event.preventDefault();
+    this.setState({
+      [event.target.id]: event.target.value,
+    });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const { title, category } = this.state;
+    const { createBook } = this.props;
+    const book = {
+      id: uuidv1(),
+      title,
+      category,
+    };
+
+    if (title && category) {
+      createBook(book);
+      this.setState({ title: '' });
+      this.title.value = '';
```

Comment:
if you set the value of the input field to `this.state.title` you would not need the above line, because once the state is empty from setState above the value becomes empty too :slightly_smiling_face: